src/extension.ts

```diff
@@ -69,7 +70,32 @@ export async function activate(context: vscode.ExtensionContext, perfStats: { lo
         activateContext.telemetry.properties.isActivationEvent = 'true';
         activateContext.telemetry.measurements.mainFileLoad = (perfStats.loadEndTime - perfStats.loadStartTime) / 1000;
 
-        ext.subscriptionProviderFactory = createVSCodeAzureSubscriptionProviderFactory();
+        // if this for a nightly test, we want to use the test subscription provider
+        const useAzureFederatedCredentials: boolean = !/^(false|0)?$/i.test(process.env['AzCode_UseAzureFederatedCredentials'] || '')
+        if (useAzureFederatedCredentials) {
+            // when running tests, ensure we throw the errors and they aren't silently swallowed
```

Comment:
Can you pull this out of the `extension.ts` code and make a new file for it? Just to help clean up our activate command.

Reply:
Done 🫡